### PR TITLE
Rename T() to tensor()

### PR DESCRIPTION
**Pull request by piscisaureus**

### CODE REVIEW

**Comment by piscisaureus** [comment]
After landing this, check that the example notebooks still work.

**Comment by piscisaureus** [comment]
TODOs after landing this:

* update the example notebooks
* re-run gendoc
* release new npm package

**Comment by piscisaureus** [comment]
Note that the tests are failing ("Params is not a constructor"), which is unrelated to this PR.
This should be fixed by re-running gendoc.
I'm not sure why the CI didn't catch this when testing #270.

File: src/api.ts [hunk 1]
```diff
@@ -163,7 +167,7 @@ export function ones(shape: types.Shape,
   return ops.ones(shape, opts);
 }
 
-interface SGDOpts {
+export interface SGDOpts {
```
Comment (ry) [inline src/api.ts]: Unrelated change?

File: website/docs.ts [hunk 2]
```diff
@@ -125,9 +125,6 @@ function startsWithUpperCase(s: string): boolean {
 export function Docs(props) {
   let docs: DocEntry[] = require("./docs.json");
   docs = docs.sort((a, b) => {
-    // Special case "T" to be at the top of the docs.
-    if (a.name === "T") return -1;
-    if (b.name === "T") return 1;
```
Comment (ry) [inline website/docs.ts]: can we leave it at the top of the docs?

Comment (piscisaureus) [reply website/docs.ts]: Of course we can, but I thought it looked kinda weird.

Comment (ry) [reply website/docs.ts]: ok nevermind
